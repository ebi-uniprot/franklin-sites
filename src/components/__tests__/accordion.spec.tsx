import { screen, render, fireEvent } from '@testing-library/react';

import Accordion from '../accordion';

const props = {
  title: 'foo',
  count: 10,
  children: <div>Foo bar baz qux quux quuz</div>,
};

describe('Accordion', () => {
  it('should render', () => {
    const { asFragment } = render(<Accordion {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should show content when expand title is clicked', () => {
    render(<Accordion {...props} />);
    const title = screen.getByTestId('accordion-title');
    fireEvent.click(title);
    const content = screen.getByTestId('accordion-content');
    expect(content).toHaveClass(
      'accordion__content',
      'accordion__content--display-content'
    );
  });

  it('should not show content when expand title is clicked twice', () => {
    render(<Accordion {...props} />);
    const title = screen.getByTestId('accordion-title');
    fireEvent.click(title);
    fireEvent.click(title);
    const content = screen.getByTestId('accordion-content');
    expect(content).toHaveClass(
      'accordion__content',
      'accordion__content--hide-content'
    );
  });
});
