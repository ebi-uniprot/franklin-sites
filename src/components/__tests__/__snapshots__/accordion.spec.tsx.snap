// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should render 1`] = `
<DocumentFragment>
  <div
    class="accordion"
  >
    <div
      class="accordion__title"
      data-testid="accordion-title"
      role="button"
      tabindex="0"
    >
      <div
        class="accordion__title__text"
      >
        foo
      </div>
      <div
        class="accordion__title__side"
      >
        <span
          class="accordion__title__side__count bubble--small"
        >
          10
        </span>
        <test-file-stub
          classname="accordion__title__side__chevron"
          height="16"
          width="16"
        />
      </div>
    </div>
    <div
      class="accordion__content accordion__content--hide-content"
      data-testid="accordion-content"
    >
      <div>
        Foo bar baz qux quux quuz
      </div>
    </div>
  </div>
</DocumentFragment>
`;
